File: src/io/innofang/singleton/example/java/SingletonManager.java
package io.innofang.singleton.example.java;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Inno Fang on 2017/8/12.
 */
public class SingletonManager {
    private static Map<String, Object> objectMap = new HashMap<>();

    private SingletonManager() {
    }

    public static void registerService(String key, Object instance) {
        if (!objectMap.containsKey(key)) {
            objectMap.put(key, instance);
        }
    }

    public static Object getService(String key) {
        return objectMap.get(key);
    }
}

File: src/io/innofang/singleton/example/kotlin/SingletonManager.kt
package io.innofang.singleton.example.kotlin

import java.util.HashMap

/**
 * Created by Inno Fang on 2017/8/12.
 */
class SingletonManager {
    private val objectMap = HashMap<String, Any>()

    private fun SingletonManager() {}

    fun registerService(key: String, instance: Any) {
        if (!objectMap.containsKey(key)) {
            objectMap.put(key, instance)
        }
    }

    fun getService(key: String): Any {
        return objectMap[key]!!
    }
}